helm/tenant/values.yaml

```diff
@@ -1,189 +1,305 @@
-## Secret with default environment variable configurations to be used by MinIO Tenant.
-## Not recommended for production deployments! Create the secret manually instead.
+###
+# Specify a Kubernetes secret with environment variables to pass to the MinIO Tenant.
+#
+# For development or PoC deployments, you can dynamically create the secret by specifying the ``name`` and then a list of environment variables. 
+# For example:
+#
+# .. code-block:: yaml
+#
+#    name: myminio-env-configuration
+#    accessKey: minio
+#    secretKey: minio123
 secrets:
-  name: myminio-env-configuration
-  # MinIO root user and password
-  accessKey: minio
-  secretKey: minio123
-  ## Set the value for existingSecret to use a pre created secret and dont create default one
-  # existingSecret: random-env-configuration
-## MinIO Tenant Definition
+  existingSecret: random-env-configuration
```

Comment:
How this functions is not clear to me. We also have `tenant.configuration`  and `tenant.extraResources`. What is the purpose specifying the secret to be created here at all, even from a development perspective?

Reply:
If you want to recreate a tenant where other used to be (ie: 1- helm install, 2-helm uninstall, 3 helm install again), Operator will recreate environment configuration secret.

This section will instruct operator to do not recreate the secret, instead use one already existing secret, created on the first `helm install`.

Sometimes is needed to uninstall helm chart for troubleshooting, or very steep upgrades

Reply:
Gotcha - in the original example we have

```yaml
## Secret with default environment variable configurations to be used by MinIO Tenant.
## Not recommended for production deployments! Create the secret manually instead.
secrets:
  name: myminio-env-configuration
  # MinIO root user and password
  accessKey: minio
  secretKey: minio123
  ## Set the value for exist
 ```

How does this compare? It looks like we exclusively use it [here](https://github.com/minio/operator/blob/master/helm/tenant/templates/tenant-configuration.yaml) and only for setting root access/secret key.